source/manual/test-and-build-a-project-with-github-actions.html.md

```diff
@@ -0,0 +1,266 @@
+      - uses: actions/cache@v1
+        with:
+          path: vendor/bundle
+          key: bundle-${{ matrix.ruby }}-${{ hashFiles('**/*.gemspec') }}
+          restore-keys: bundle-${{ matrix.ruby }}
+      - run: bundle install --jobs 4 --retry 3 --path vendor/bundle
+      - run: bundle exec rake
+```
+
+Notes:
+
+- The [deployment][bundler-deployment] flag for bundler cannot be used for gems
+  as they should not have a Gemfile.lock committed.
+- Our preference is to test the gems we publish against the latest version of
+  [all currently supported minor versions of Ruby MRI][ruby-branches].
+
+## GOV.UK Rails application with Postgres, Redis, Yarn and GOV.UK Content Schemas dependencies
+
+This is an example that suits a relatively complex GOV.UK Rails application
+(this was written for [Content Publisher][]) that has software dependencies
+(provided by [Docker containers][actions-docker-services]), cloning of a
+supplementary repository, and the use of Node.js for JavaScript dependency
+management.
+
+```yml
+name: Continuous Integration
+on: [push, pull_request]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:9.4
+        ports:
+          - 5432:5432
+        env:
+          POSTGRES_HOST_AUTH_METHOD: trust
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+      redis:
+        image: redis
+        ports:
+          - 6379:6379
+        options: --health-cmd "redis-cli ping" --health-interval 10s --health-timeout 5s --health-retries 5
+    env:
+      RAILS_ENV: test
+      REDIS_URL: redis://localhost:6379/0
+      TEST_DATABASE_URL: postgresql://postgres@localhost/content-publisher
+      GOVUK_CONTENT_SCHEMAS_PATH: vendor/govuk-content-schemas
+    steps:
+      - name: Clone project
+        uses: actions/checkout@v2
+      - name: Clone GOV.UK Content Schemas
+        uses: actions/checkout@v2
+        with:
+          repository: alphagov/govuk-content-schemas
+          ref: deployed-to-production
+          path: vendor/govuk-content-schemas
+      - uses: ruby/setup-ruby@v1
+      - name: Check for cached bundler
+        uses: actions/cache@v1
+        with:
+          path: vendor/bundle
+          key: bundle-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: bundle
+      - run: bundle install --jobs 4 --retry 3 --deployment
+      - name: Install Node for JS dependencies
+        uses: actions/setup-node@v1
+      - run: npm install -g yarn
+      - name: Check for cached node modules
+        uses: actions/cache@v1
+        with:
+          path: node_modules
+          key: yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: yarn
+      - run: yarn install --frozen-lockfile
+      - run: bundle exec rails db:setup
+      - run: bundle exec rails assets:precompile
+      - run: bundle exec rake
+```
+
+Notes:
+
+- Similar configuration to PostgreSQL and Redis can be applied to other software
```

Comment:
Minor: how would this help someone reading it?

Reply:
Mostly this is to discourage adding more examples for every variation of a db system to this doc. Ideally it'd be good to have links to db configurations that could be used but it wasn't as easy to find them as I thought.

Reply:
Ah, I see 👍.